product_listings_manager/products.py

```diff
@@ -335,3 +418,54 @@ def getProductListings(productLabel, buildInfo):
                 if len(maps) == 1 and maps[0] == 'src' and not allow_src_only:
                     del listings[variant]
     return listings
+
+
+def getModuleProductListings(productLabel, moduleInfo):
+    """
+    Get a map of which variants of the given product included the given module,
+    and which arches each variant included.
```

Comment:
It would be really helpful to be explicit about what the expected format of `moduleInfo` is. As far as I can tell from the code, it is the NVR of the module build as Koji understands it.

It's important to note that Koji's NVR is not really NSVC. According to modulemd spec, NSVC is delimited by `:` characters. When importing to Koji, MBS transforms `N:S:V:C` to `N-S'-V.C` (changing the delimiters, and replacing any dashes in stream with underscores).

As far as I know, ET is using the NVR representation, so the selected value is a good one, but I would like to see it written here explicitly.

Reply:
> The format is slightly different to what is returned from `getProductListings`. That call has one two more levels: it includes package name (to accommodate for subpackages) and package architecture (to accommodate for multilib). Neither of those concepts are applicable to modules, so it should be fine to not have it for module listings. It would however be good to get feedback from someone from ET to verify this is going to work there. @hluk maybe?

The format is based on this comment https://projects.engineering.redhat.com/browse/RCMWORK-9850?focusedCommentId=1061402&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-1061402

Reply:
> It would be really helpful to be explicit about what the expected format of `moduleInfo` is. As far as I can tell from the code, it is the NVR of the module build as Koji understands it.
> 
> It's important to note that Koji's NVR is not really NSVC. According to modulemd spec, NSVC is delimited by `:` characters. When importing to Koji, MBS transforms `N:S:V:C` to `N-S'-V.C` (changing the delimiters, and replacing any dashes in stream with underscores).
> 
> As far as I know, ET is using the NVR representation, so the selected value is a good one, but I would like to see it written here explicitly.

Got it and will fix it tmr.

Reply:
Ah, I didn't remember that comment. It looks good to me then.